### DISCUSSION TITLE
Getting direct links to S3

### DISCUSSION THREAD

**Discussion by Demoevery** [pocketbase/pocketbase#2191] [Q&A] [answered]
I was wondering if there is a way to create links so that client can recieve files directly from s3?
![drawing](https://user-images.githubusercontent.com/95992350/229276452-351cc696-e449-4670-bb38-9271c20d8ba7.png)

**Comment by ganigeorgiev** [comment, marked as answer]
In order to be able to retrieve files directly from a S3 storage you have to make your bucket public but please note that not every S3 provider support specifying bucket permissions.

The objects key/prefix is in the format `collectionId/recordId/filename`.

_You can also check https://github.com/pocketbase/pocketbase/issues/28#issuecomment-1492516269._

**Reply by marrrcin** [reply]
hi @ganigeorgiev - do you think that `pocketbase` could support serving files from S3-compatible providers by acting as a reverse proxy (e.g. with the use of `httputil.ReverseProxy`) instead of downloading files to the server and then sending them to the user?

**Reply by ganigeorgiev** [reply]
I'm not sure that I understand how this proxy will work. If we want to support private S3 buckets we have to authenticate with the S3 apis and read at least a portion of the blob before serving it (note that we support `Range` requests) and I don't think we can get away with that.

The only way I can think of allowing a private S3 bucket to actually serves the content directly is maybe to generate signed url and send it back to the client to download the file from it, but not all S3 providers support that and will complicate further the already messy filesystem abstraction.

**Reply by marrrcin** [reply]
It's actually a simpler case - just wanted to pass-through the requests to the bucket (with possibly public access) to skip the downloading to the pocketbase memory and then sending it to the client 🤔

**Reply by marrrcin** [reply]
WDYT @ganigeorgiev ?

**Reply by ganigeorgiev** [reply]
@marrrcin Again, I'm not sure that I understand how this will work in the general case and how you can skip downloading portion of the blob on the PocketBase server when proxying requests like this.

If you think that this can be implemented easily, feel free to open a PR and I'll try to review it, but for now it is not clear to me how this will work and remains out of the scope.

**Reply by abhi-94** [reply]
hello @marrrcin , can you get any way to make publicly accessible for those objects of the s3 that uploaded by the pocketbase. any information helps a lop man.
